mlflow/deployments/base.py

```diff
@@ -190,3 +191,20 @@ def predict(self, deployment_name, df):
         :return: A pandas DataFrame, pandas Series, or numpy array
         """
         pass
+
+    @experimental
+    def explain(self, deployment_name, df):  # pylint: disable=unused-argument
+        """
+        Generate explanations of model predictions on the specified input pandas Dataframe
+        ``df`` for the deployed model. Explanation output formats vary by deployment target,
+        and can include details like feature importance for understanding/debugging predictions.
+        Note that we have created this function as non-abstract method to not break
+        the existing downstream.
+
+        :param deployment_name: Name of deployment to predict against
+        :param df: Pandas DataFrame to use for explaining feature importance in model prediction
+        :return: An exception if the implementation is not available in deployment target's class
```

Comment:
Could we clarify that the expected return type is a JSON-able object, e.g. a pandas dataframe or numpy array? 
```suggestion
        :return: A JSON-able object (pandas dataframe, numpy array, dictionary), or exception if unimplemented for the current deployment target
```